registry/txt.go

```diff
@@ -195,8 +195,13 @@ func (im *TXTRegistry) generateTXTRecord(r *endpoint.Endpoint) []*endpoint.Endpo
 	txt := endpoint.NewEndpoint(im.mapper.toTXTName(r.DNSName), endpoint.RecordTypeTXT, r.Labels.Serialize(true)).WithSetIdentifier(r.SetIdentifier)
 	txt.ProviderSpecific = r.ProviderSpecific
 	// new TXT record format (containing record type)
-	txtNew := endpoint.NewEndpoint(im.mapper.toNewTXTName(r.DNSName, r.RecordType), endpoint.RecordTypeTXT, r.Labels.Serialize(true)).WithSetIdentifier(r.SetIdentifier)
-	txtNew.ProviderSpecific = r.ProviderSpecific
+	txtNew := endpoint.NewEndpoint(im.mapper.toNewTXTName(r.DNSName, r.RecordType), endpoint.RecordTypeTXT, r.Labels.Serialize(true))
+	if txtNew != nil {
+		txtNew.WithSetIdentifier(r.SetIdentifier)
+		txtNew.ProviderSpecific = r.ProviderSpecific
+	} else {
+		return []*endpoint.Endpoint{txt}
```

Comment:
should we warn about this fallback is used?
I am not completely sure anymore about what behavior this will introduce in the long run.
I guess we loose only the ability to create other records of the same name, which seems ok for these kind of long records.

Reply:
Not using a subdomain for the prefix was a mistake, I think. And I guess that is the migration path for those that need e.g AAAA-records.

Reply:
We already warn in `NewEndpoingWithTTL`. Warning again is perhaps redundant?

Reply:
Ah yes, you are right with the return nil it's the same case :)
Subdomain can be used as far as I see by using --txt-prefix or --txt-suffix or example similar to what was written in https://github.com/kubernetes-sigs/external-dns/issues/2839#issuecomment-1205025159.

Reply:
/approve
/lgtm